pkg/util/docker/filter_test.go

```diff
@@ -105,16 +105,41 @@ func TestContainerFilter(t *testing.T) {
 		},
 	} {
 		t.Run("", func(t *testing.T) {
-			f, err := newContainerFilter(tc.whitelist, tc.blacklist)
+			f, err := NewFilter(tc.whitelist, tc.blacklist)
 			require.Nil(t, err, "case %d", i)
 
 			var allowed []string
 			for _, c := range containers {
-				if !f.IsExcluded(c) {
+				if !f.IsExcluded(c.Name, c.Image) {
 					allowed = append(allowed, c.ID)
 				}
 			}
 			assert.Equal(t, tc.expectedIDs, allowed, "case %d", i)
 		})
 	}
 }
+
+// NewFilterFromConfig creates a new container filter, sourcing patterns
+// from the pkg/config options
+func TestNewFilterFromConfig(t *testing.T) {
+	config.Datadog.SetDefault("exclude_pause_container", true)
+	config.Datadog.SetDefault("ac_include", []string{"image:apache.*"})
+	config.Datadog.SetDefault("ac_exclude", []string{"name:dd-.*"})
+
+	f, err := NewFilterFromConfig()
+	require.NoError(t, err)
+
+	assert.True(t, f.IsExcluded("dd-152462", "dummy:latest"))
+	assert.False(t, f.IsExcluded("dd-152462", "apache:latest"))
+	assert.False(t, f.IsExcluded("dummy", "dummy"))
+	assert.True(t, f.IsExcluded("dummy", "k8s.gcr.io/pause-amd64:3.1"))
+
+	config.Datadog.SetDefault("exclude_pause_container", false)
+	f, err = NewFilterFromConfig()
+	require.NoError(t, err)
+	assert.False(t, f.IsExcluded("dummy", "k8s.gcr.io/pause-amd64:3.1"))
+
+	config.Datadog.SetDefault("exclude_pause_container", true)
```

Comment:
nits, 
Is it for a tearDown ?

In that case, it should be:
* defer
* tearDown in a suite
* remove any requires